packages/ncids-js/src/components/nci-header/utils/mobile-menu/mobile-menu.ts

```diff
@@ -0,0 +1,457 @@
+			const results = await this.adaptor.getNavigationLevel(initialMenuId);
+			const menu = this.displayNavLevel(results);
+			this.mobileNav.append(menu);
+
+			this.focusTrap.toggleTrap(true, this.mobileNav);
+			this.element.dispatchEvent(this.customEvents['open']);
+		}
+	}
+
+	/**
+	 * Handle Link Click
+	 * @private
+	 */
+	private async handleClick(event: Event): Promise<void> {
+		event.preventDefault();
+		const link = <HTMLElement>event.target;
+		const path = this.adaptor.useUrlForNavigationId
+			? <string>link.getAttribute('data-href')
+			: <string>link.getAttribute('data-menu-id');
+
+		this.lastClicked = path;
+
+		const results = await this.adaptor.getNavigationLevel(path);
+		const menu = this.displayNavLevel(results);
+		this.mobileNav.append(menu);
+
+		this.focusTrap.toggleTrap(true, this.mobileNav);
+	}
+
+	/**
+	 * Display the current level of navigation
+	 *
+	 * @param {MobileMenuItem[]} data child subtree.
+	 * @param {MobileMenuItem} parent parent item.
+	 *
+	 * @returns {HTMLElement}
+	 * @private
+	 */
+	private displayNavLevel(data: MobileMenuData): HTMLElement {
+		const items = data.items;
+		const parent = data.parent;
+		if (parent?.path) this.sectionParent = parent.path;
+		// if we have a menu already - then remove it and redraw
+		const menuCheck = this.element.querySelector('.nci-header-mobilenav__list');
+		if (menuCheck) menuCheck.remove();
+
+		const menuList = this.createDom('ul', ['nci-header-mobilenav__list']);
+		const menuParent = this.createDom('ul', ['nci-header-mobilenav__list']);
+
+		// if the parent item in our return data is not a root path
+		if (parent !== null) {
+			const menuHolder = this.createDom('li', [
+				'nci-header-mobilenav__list-holder',
+			]);
+
+			// make parent back to menu link
+			const menuBack = this.makeBackNode(parent);
+			const exploreSection = this.makeExploreLink(data);
+			menuParent.append(menuBack);
+			menuList.append(exploreSection);
+
+			menuHolder.append(menuList);
+			menuParent.append(menuHolder);
+
+			items.map((item) => {
+				let mobileItem;
+				if (item.hasChildren) {
+					mobileItem = this.makeMenuNode(item);
+				} else {
+					mobileItem = this.makeMenuLink(item);
+				}
+				menuList.append(mobileItem);
+			});
+		} else {
+			// else it's the default menu
+			items.map((item) => {
+				let mobileItem;
+				if (item.hasChildren) {
+					mobileItem = this.makeMenuNode(item);
+				} else {
+					mobileItem = this.makeMenuLink(item);
+				}
+				menuList.append(mobileItem);
+			});
+		}
+
+		// return parent or submenu base on path
+		if (data.path !== '/') {
+			return <HTMLElement>menuParent;
+		} else {
+			return <HTMLElement>menuList;
+		}
+	}
+
+	/**
+	 * Create Back Button for current section menu
+	 *
+	 * @returns {HTMLElement}
+	 * @private
+	 */
+	private makeBackNode(item: MobileMenuItem): HTMLElement {
+		let label;
+		let path;
+		const lang = this.langCode;
+		if (this.sectionParent === undefined) {
+			path = '/';
```

Comment:
Looks like there is missing code coverage for the first part of the if statement. Second half looks covered?